packages/credential-provider-imds/src/utils/staticStabilityProvider.ts

```diff
@@ -32,7 +33,7 @@ export const staticStabilityProvider = (
         credentials = getExtendedInstanceMetadataCredentials(credentials, logger);
       }
     } catch (e) {
-      if (pastCredentials) {
+      if (pastCredentials && !(e instanceof InstanceMetadataV1FallbackError)) {
```

Comment:
Clarifying question: so if the error is `InstanceMetadataV1FallbackError`, it shouldn't try to extend the credentials?

Reply:
yeah, it should still extend, so I'll remove this condition